static/js/attachments_ui.js

```diff
@@ -33,9 +19,45 @@ function bytes_to_size(bytes, kb_with_1024_bytes) {
         size = Math.round(bytes / Math.pow(kb_size, i) * 10) / 10;
     }
     return size + ' ' + sizes[i];
+};
+
+exports.percentage_used_space = function (uploads_size) {
+    if (page_params.realm_upload_quota === null) {
+        return null;
+    }
+    return (uploads_size / page_params.realm_upload_quota).toFixed(1);
+};
+
+exports.set_upload_space_stats = function () {
+    channel.get({
+        url: "/json/realm/upload_space_used",
+        idempotent: true,
+        success: function (data) {
+            var percent_used = exports.percentage_used_space(data.upload_space_used);
+            var upload_quota = attachments_ui.bytes_to_size(page_params.realm_upload_quota, true);
+            $("#upload_percentage_used").text(percent_used);
+            $("#upload_quota").text(upload_quota);
```

Comment:
So, the problem with this approach is that there's a brief glitch where the data isn't displayed yet.  We should instead have a small handlebars template for this widget, and just have the whole element added by this code path.

Reply:
Alternatively, there's a `channel.get` call that we already do before rendering this UI (see below); maybe we should just add the `upload_space_used` value to the dictionary returned there (removing the new endpoint) and remove this extra call?  That might be cleaner.